src/hooks/useCourtContracts.js

```diff
@@ -494,25 +520,27 @@ export function useTotalActiveBalancePolling(termId) {
   )
   const [totalActiveBalance, setTotalActiveBalance] = useState(bigNum(-1))
 
-  const timeoutId = useRef(null)
-
   useEffect(() => {
     let cancelled = false
+    let timeoutId
 
     const fetchTotalActiveBalance = () => {
-      timeoutId.current = setTimeout(() => {
+      timeoutId = setTimeout(() => {
         return jurorRegistryContract
           .totalActiveBalanceAt(termId)
           .then(balance => {
             if (!cancelled) {
               setTotalActiveBalance(balance)
-              clearTimeout(timeoutId.current)
-              fetchTotalActiveBalance()
             }
           })
           .catch(err => {
             console.error(`Error fetching balance: ${err} retrying...`)
-            fetchTotalActiveBalance()
+          })
+          .finally(() => {
+            if (!cancelled) {
+              clearTimeout(timeoutId)
```

Comment:
I wonder if we need to do this; the related timer should have already been finished at this point (it is what invokes this function)?

Reply:
Ah yes good point, it was needed in the prior implementation but indeed we don't need it now.